swift/apple/FirezoneKit/Sources/FirezoneKit/Views/MenuBar.swift

```diff
@@ -422,7 +422,9 @@ public final class MenuBar: NSObject, ObservableObject {
     }
 
     menu.addItem(aboutMenuItem)
-    menu.addItem(adminPortalMenuItem)
+    if !(store.configuration?.hideAdminPortalMenuItem ?? false) {
```

Comment:
It feels very brittle that we have so many null-checks and default here. Wouldn't it be better to do all of this in the `Store` or at least always initialise the configuration and do it on there?

Reply:
Yeah, we could just have computed properties to return defaults for these. I was trying to maintain the third state of "not set" everywhere because I don't know where we'll use it, but having the default applied on the read will allow that.